src/eva/data/mon_data_space.py

```diff
@@ -0,0 +1,302 @@
+            #         has a problem.  The get() function returns a variety of types:
+            #
+            #         yaml file input                   return value   type
+            #         ---------------                   ------------   ----
+            #  	      regions: &regions 2,4             2,4            str
+            #         regions: &regions 1               1              int
+            #         regions: &regions                 None           None
+            #         regions: &regions all	        ---            ---   execution aborts
+            #         regions: &regions [2,4]           ---            ---   execution aborts
+            #
+            #         Is the variable return type a bug or feature?   And note that the last 2 are valid
+            #         when parcing the desired variable values.  This behavior should be consistent, no?
+            #
+            #         Also worth noting -- a request for region without a region entry in the 
+            #         the yaml file causes an abort.  I guess that's ok, but it seems like no value should 
+            #         be equivalent to all.
+
+            requested_regions = []
+            drop_regions = False
+
+            if regions_str_or_list is not None: 
+                self.logger.info(' inside not None')
+
+                if len(str(regions_str_or_list)) > 0:
+                    requested_regions = parse_channel_list(str(regions_str_or_list), self.logger)
+                    drop_regions = True
+
+            # Filenames to be read into this collection
+            # -----------------------------------------
+            filenames = get(dataset, self.logger, 'filenames')
+            individual_files = []
+
+            for filename in filenames:
+                count_tmp, penalty_tmp, omgnbc_sum_tmp, total_sum_tmp, \
+                omgbc_sum_tmp, omgnbc_sum2_tmp, total_sum2_tmp, omgbc_sum2_tmp, cycle_tm = \
+                self.read_radmon_ieee(filename, nchans, nregion)
+
+                timestep_ds = Dataset(
+                    {
+                        "count": (("channels", "regions"), count_tmp),
+                        "penalty": (("channels", "regions"), penalty_tmp),
+                        "omgnbc": (("channels", "regions"), omgnbc_sum_tmp),
+                        "total": (("channels", "regions"), total_sum_tmp),
+                        "omgbc": (("channels", "regions"), omgbc_sum_tmp),
+                        "omgnbc2": (("channels", "regions"), omgnbc_sum2_tmp),
+                        "total2": (("channels", "regions"), total_sum2_tmp),
+                        "omgbc2": (("channels", "regions"), omgbc_sum2_tmp),
+                    },
+                    coords={"channels": channo, "regions": np.arange(1,nregion+1)},
+                    attrs = {'satellite': satellite, 'sensor': sensor},
+                )
+                timestep_ds['times'] = cycle_tm
```

Comment:
I'm not sure I'm on the right track here.  The idea is to read in n data files, as specified in the testMonDataSpaceHirs4Metop-A.yaml file, and then concatenate the dataset arrays in individual_files by timestep.  That works fine, but I don't know how to specify the time as the x axis when requesting a time series plot.  Time here is a coordinate not a variable.  I'll post the output of the final dataset below.  I'd be glad to learn what I'm doing wrong.

`INFO MonDataSpace: ds: <xarray.Dataset>
INFO MonDataSpace: Dimensions:   (times: 5, channels: 1, regions: 1)
INFO MonDataSpace: Coordinates:
INFO MonDataSpace:   * channels  (channels) int64 1
INFO MonDataSpace:   * regions   (regions) int64 1
INFO MonDataSpace:   * times     (times) int64 2015051418 2015051500 ... 2015051512 2015051518
INFO MonDataSpace: Data variables:
INFO MonDataSpace:     count     (times, channels, regions) float32 9.065e+03 ... 9.01e+03
INFO MonDataSpace: Attributes:
INFO MonDataSpace:     satellite:  metop-a
INFO MonDataSpace:     sensor:     hirs4
`

Reply:
I tried a brute force hack to add a variable of cycle_time to the ds.  It sort of works; the counts are plausible, but the x label and increments make little sense:

![hirs4_metop-a 0 count](https://user-images.githubusercontent.com/62339196/219452564-d5ccddc7-6fbb-4595-bd9b-8652e45fbabb.png)

Reply:
Now with a datetime object instead it looks much better:

![hirs4_metop-a 0 count](https://user-images.githubusercontent.com/62339196/219712300-ee8c7a69-5aee-4dfa-a0ae-eedd1bfffd1f.png)

Reply:
Now with a datetime object instead it looks much better:

![hirs4_metop-a 0 count](https://user-images.githubusercontent.com/62339196/219712300-ee8c7a69-5aee-4dfa-a0ae-eedd1bfffd1f.png)